### ISSUE TITLE
CDa for Edge Labeled Graphs

### ISSUE THREAD

**Issue by rudra0713** [GiulioRossetti/cdlib#218]
Hi,
   I am working with a graph that has both node attributes (numerical) and edge labels (categorical, string-like). I checked the community detection algorithms listed in 
https://cdlib.readthedocs.io/en/latest/reference/cd_algorithms/node_clustering.html, 
and saw that `eva` and `ilouvain` algorithms are capable of handling node attributes. However, from the descriptions, I am not finding an algorithm that can handle edge-labeled graphs. 

Is there any CD algorithm implemented in cdlib that can handle both labeled-edge and node attributes?

**Comment by Yquetzal** [comment]
No, I don't think so. Do you know examples of such algorithms ?

**Comment by rudra0713** [comment]
Thanks for your comment @Yquetzal. Just to double-check, edge labels can simply be a color for each edge from a set of colors. For example, each edge can be colored blue, red, or green. (This is just a toy example.)

I recently found these two works that focus on CD for edge-labeled graphs.

http://www.cs.cmu.edu/~maraujo/papers/mlgworkshop16.pdf
https://arxiv.org/pdf/2210.10583.pdf
